src/main/java/com/salesforce/mirus/MirusSourceTask.java

```diff
@@ -197,6 +206,35 @@ private void seekToOffsets(List<TopicPartition> partitionIds) {
     return Collections.emptyList();
   }
 
+  @Override
+  public void commit() {
+    setCommitTime();
+  }
+
+  private void setPollTime() {
+    long currentPollTime = time.milliseconds();
+    // no new data in the commit window, reset commit time
+    if ((currentPollTime - pollTime) >= commitFailureRestartMs) {
+      setCommitTime();
+    }
+    pollTime = currentPollTime;
+  }
+
+  private void setCommitTime() {
+    commitTime = time.milliseconds();
+  }
+
+  private void checkCommitFailure() {
+    // if no success offset commit in an extensive period of time, restart task to reestablish Kafka
+    // connection
+    long noCommitMs = pollTime - commitTime;
+    if (noCommitMs >= commitFailureRestartMs) {
+      logger.error(
```

Comment:
The exception will generate a log line, so maybe merge the log message into the exception message and drop the `log.error()` line?